kernel/src/process_loading.rs

```diff
@@ -869,13 +938,466 @@ impl<C: Chip, D: ProcessStandardDebug> SequentialProcessLoaderMachine<'_, C, D>
 
         blocks
     }
+
+    ////////////////////////////////////////////////////////////////////////////////
+    // DYNAMIC PROCESS LOADING HELPERS
+    ////////////////////////////////////////////////////////////////////////////////
+    /// TODO:
+    /// 1. Block app from writing if the version is same and if the binary has not
+    ///     changed? (possible attack vector)
+    ///         - Change version and keep binary image constant otherwise.
+    ///         - A bunch of these updates could eat away at the flash's real estate
+
+    /// Scan the entire flash to populate lists of existing binaries addresses.
+    fn scan_flash_for_app_binaries(
+        &self,
+        flash: &'static [u8],
+        process_binaries_start_addresses: &mut [usize; MAX_PROCS],
+        process_binaries_end_addresses: &mut [usize; MAX_PROCS],
+    ) -> Result<(), ProcessBinaryError> {
```

Comment:
This is confusing that this function is "supposed" to fail. This should return `Result<(), ()>` probably, and the normal case is the arrays are populated correctly.

To make this easier, you can define a function within this function that does return `Result<(), ProcessBinaryError>` so the `?` still works, and then essentially call that inner function and fixup the return value.